### PR TITLE
Patch in ps2/opl support for game scanning

### LINKED ISSUE #324: [Feature] PS2/OPL support
**Is your feature request related to a problem? Please describe.**
no, just something that would be nicer to have.

**Describe the solution you'd like**
opl or open ps2 loader loads iso from a nas or network share.
it has a 32 character limit. meaning some of the name is cut off
it also doesnt require the SERIAL to be in the name but loads faster with them in the name.
```
SLUS_207.45.Shrek 2.iso
SLUS_206.85.Ape Escape 2.iso
```
it be nice if you could scan the iso for the SERIAL there are a few tools out there that do this very thing. the SERIAL then could be used on a list (which are out there as well) to find the full name.
**Describe alternatives you've considered**
the ability to skip over the SERIAL  in the filename
```
SLUS_207.45.Shrek 2.iso
SLUS_206.85.Ape Escape 2.iso

```
becomes 
```
Shrek 2.iso
Ape Escape 2.iso
```
**Additional context**
this sound only need to happen with ps2 games.
ps2 iso renamer: https://www.psx-place.com/resources/eziso-renamer-by-raydenx.700/
another ps2 renamer: http://www.psxrenzukoken.com/nk2/download/psx-renamer-1-7-database/
this also has a database file the could be used.

### PR DESCRIPTION
**Pull request by GAntoine**
Patches in support for the [OPL](https://github.com/ps2homebrew/Open-PS2-Loader) file naming convention. In the case where the filename start with a value matching the OPL serial regex (XXXX_000.00), use the name from the index to search IGDB. This also has the added benefit of using that name over the filename when the IGDB lookup fails.

Note: I chose to implement the index as an in-memory python dict due to it's relatively small size (1.1MB), and not having to load it in from a fixture file (JSON, YAML). In the future, objects like these can be stored in Redis to offload that memory footprint, while allowing for fast-ish access. Since you can't view it in github, here's the structure of that index:

```python
opl_index = {
  "SLES_556.71": {
    "Name": "Fifa '14",
    "Region": "PAL"
  },
  "SLES_556.72": {
    "Name": "Fifa '14",
    "Region": "PAL"
  },
 ...
}
```

* Also fixes a small bug with scan result reporting in client

cc @binarygeek119 if you want to have a look

Closes #324 

Screenshot 2023-08-08 at 5 34 17 PM

### CODE REVIEW

File: backend/handler/igdb_handler.py [hunk 1]
```diff
@@ -114,11 +116,20 @@ def get_platform(self, p_slug: str):
...
-        search_term = uc(get_search_term(file_name))
+        search_term = get_search_term(file_name)
+
+        # Patch support for PS2 OPL filename format
+        match = re.match(ps2_opl_regex, search_term)
+        if p_igdb_id == 8 and match:
+            serial_code = match.group(1)
+            index_entry = opl_index.get(serial_code, None)
+            if index_entry:
+                search_term = index_entry["Name"]
```
Comment (gantoine) [inline backend/handler/igdb_handler.py:133]: The change is very self-contained, so as not to pollute the rest of the code base.

**Review by zurdi15 — APPROVED** [review]

**Comment by binarygeek119** [comment]
i just tryed to load a ps2 game and got this error.
my game is named 1 but is 4x4 Evo (USA).iso
```
INFO:     [RomM][2023-08-31 05:06:34]   Identified as PlayStation 2 🎮
INFO:     [RomM][2023-08-31 05:06:34]    · SLUS_200.91.1.iso
Task exception was never retrieved
future: <Task finished name='Task-117' coro=<AsyncServer._handle_event_internal() done, defined at /usr/local/lib/python3.10/dist-packages/socketio/asyncio_server.py:522> exception=FileNotFoundError(2, 'No such file or directory')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/socketio/asyncio_server.py", line 524, in _handle_event_internal
    r = await server._trigger_event(data[0], namespace, sid, *data[1:])
  File "/usr/local/lib/python3.10/dist-packages/socketio/asyncio_server.py", line 558, in _trigger_event
    ret = await handler(*args)
  File "/back/endpoints/scan.py", line 97, in scan_handler
    await scan_platforms(platform_slugs, complete_rescan, selected_roms)
  File "/back/endpoints/scan.py", line 60, in scan_platforms
    scanned_rom = fastapi.scan_rom(scanned_platform, fs_rom)
  File "/back/utils/fastapi.py", line 60, in scan_rom
    file_size, file_size_units = fs.get_rom_size(
  File "/back/utils/fs.py", line 262, in get_rom_size
    total_size += os.stat(file).st_size
FileNotFoundError: [Errno 2] No such file or directory: '/romm/library/roms/ps2/SLUS_200.91.1.iso'
```

**Comment by GAntoine** [comment]
What's the exact file name of the game in the filesystem?

**Comment by binarygeek119** [comment]
figured it out seems romm doesnt like symbolic link. is there a way of making it happy with symbolic links i need the files to be in two places at the sametime with out taking up double the space.

**Comment by GAntoine** [comment]
How are you creating the symlink, `ln -s`?

**Comment by binarygeek119** [comment]
ln -sf source destination
is a batch file to regenerate when new files are added

**Comment by GAntoine** [comment]
> i need the files to be in two places at the same time with out taking up double the space

If this is on the same disk, you should use a hardlink instead so they're all pointing to the same inode: `ln source destination`

**Comment by binarygeek119** [comment]
thats did it thanks

**Pull request merged**